EventGenerator/src/CORSIKAEventGenerator_module.cc

```diff
@@ -59,11 +60,12 @@ namespace mu2e {
         using Name = fhicl::Name;
         using Comment = fhicl::Comment;
 
-        fhicl::Atom<std::string> corsikaModuleLabel{Name("corsikaModuleLabel"), Comment("Reference point in the coordinate system"), "FromCorsikaBinary"};
+        fhicl::Atom<std::string> corsikaModuleLabel{Name("corsikaModuleLabel"), Comment("Name of CORSIKA module label"), "FromCorsikaBinary"};
         fhicl::Atom<std::string> refPointChoice{Name("refPointChoice"), Comment("Reference point in the coordinate system"), "UNDEFINED"};
         fhicl::Atom<bool> projectToTargetBox{Name("projectToTargetBox"), Comment("Store only events that cross the target box"), false};
         fhicl::Atom<float> targetBoxYmax{Name("targetBoxYmax"), Comment("Top coordinate of the target box on the Y axis")};
         fhicl::Atom<float> intDist{Name("intDist"), Comment("Maximum distance of closest approach for backtracked trajectories")};
+        fhicl::Atom<art::InputTag> simTimeOffset{Name("simTimeOffset"), Comment("Simulation time offset to apply (optional)"), art::InputTag()};
```

Comment:
This should be OptionalAtom if it is optional.  Then you can test on it being set instead of being null.

Reply:
I changed it here and also in `Mu2eProductMixer`.